### ISSUE TITLE
Onedrive Listener typeError Exception

### ISSUE THREAD

**Issue by dsturge01** [BC-SECURITY/Empire#40]
## Empire Version
3.0.3

## OS Information (Linux flavor, Python version)
5.3.0-kali3-amd64 #1 SMP Debian 5.3.15-1kali1 (2019-12-09) x86_64 GNU/Linux
Error observed on 2, 2.7, 3, 3.7

## Expected behavior and description of the error, including any actions taken immediately prior to the error. The more detail the better.
Set ClientID, ClientSecret, Authcode, and execute. The listener runs, but then throws the below exception

## Screenshot of error, embedded text output, or Pastebin link to the error
(Empire: listeners/onedrive) > execute
[*] Starting listener 'onedrive'
[*] Got new auth token
[*] empire folder already exists
[*] empire/staging already exists
[*] empire/taskings already exists
[*] empire/results already exists
[+] Listener successfully started!
(Empire: listeners/onedrive) > Exception in thread Thread-2:
Traceback (most recent call last):
  File "/usr/lib/python3.7/threading.py", line 926, in _bootstrap_inner
    self.run()
  File "/root/WD_Hacking/Empire/lib/common/helpers.py", line 983, in __run
    self.__run_backup()
  File "/usr/lib/python3.7/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/root/WD_Hacking/Empire//lib/listeners/onedrive.py", line 628, in start_server
    upload_stager()
  File "/root/WD_Hacking/Empire//lib/listeners/onedrive.py", line 564, in upload_stager
    ps_stager = self.generate_stager(listenerOptions=listener_options, language='powershell', token=token['access_token'])
  File "/root/WD_Hacking/Empire//lib/listeners/onedrive.py", line 312, in generate_stager
    return RC4IV + encryption.rc4(RC4IV+staging_key, randomized_stager)
TypeError: can't concat str to bytes

**Comment by dsturge01** [comment]
@Cx01N 
I was able to get around the issue by editing/adding the following lines to onedrive.py:

staging_key = staging_key.encode('UTF-8')

return RC4IV + encryption.rc4(RC4IV+staging_key, randomized_stager.encode('UTF-8'))

This caused the listener to run without incident, however I do not know if this will mess up encryption/communication with the agent.

---------------------------
Now when I run the agent, I get the following error:
[!] handle_agent_request(): sessionID b'YRS18TNE' not present

This error is thrown about every 5 seconds, for 5 minutes. 

Then, after about 5 minutes, the error changes to: 
[+] Initial agent YRS18TNE from 0.0.0.0 now active (Slack)
[!] Could not handle agent staging for listener onedrive, continuing
[!] handle_agent_request(): sessionID b'YRS18TNE' not present

I do not know if these are symptoms the of changes I made to onedrive.py, or if these are separate issues that needs to be opened.
